### DISCUSSION TITLE
undefined symbol: set_com_err_gettext

### DISCUSSION THREAD

**Discussion by jshen28** [pyinstaller/pyinstaller#8939] [PyInstaller] [answered]
Hello,

I am building an executable on centos 6.1 (ancient indeed) and in the python script using Pyinstaller versioned 3.2, I need to execute chattr command line to update some files permission. This works well on centos 6.1 but does not work if I move the executable to Ubuntu 18.04. And the error message is 

```
        linux-vdso.so.1 (0x00007fffce733000)
        libe2p.so.2 => /lib/x86_64-linux-gnu/libe2p.so.2 (0x00007f90f4a30000)
        libcom_err.so.2 => /tmp/_MEIiQOCIU/libcom_err.so.2 (0x00007f90f482c000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f90f463a000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f90f4617000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f90f4a45000)
chattr: symbol lookup error: chattr: undefined symbol: set_com_err_gettext
```

I guess the problem is that libcom_err.so.2  is packaged and chattr is from host, so there is a mismatch between shared library and binary file.

My question what is the recommendation if I would like to run some host commands in python script? Thank you.

Best Regards,

Jiatong Shen

**Comment by rokm** [comment, marked as answer]
> My question what is the recommendation if I would like to run some host commands in python script? Thank you.

See https://github.com/orgs/pyinstaller/discussions/8916#discussioncomment-11555218

You should either reset `LD_LIBRARY_PATH` in the environment passed to the subprocess, or at least remove the paths anchored to `sys._MEIPASS` (the top-level application directory of the frozen program) from it.
